src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/CloneDetectionAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import java.math.BigInteger
+import java.security.MessageDigest
+
+class CloneDetectionAdapter {
+    private val psiProvider = PsiProvider()
+
+    fun format(function: KtNamedFunction, projectId: Int, methodId: Int): String {
+        psiProvider.deleteComments(function)
+        val tokens: List<String> = tokenize(function.text)
+        val totalTokens = tokens.size
+        val tokensCounter: Map<String, Int> = tokens.groupingBy { it }
+            .eachCount()
+        val uniqueTokens = tokensCounter.keys.size
+        val hash: String = md5Hash(function.text)
+
+        val statsResult = listOf(
+            projectId.toString(),
+            methodId.toString(),
+            totalTokens.toString(),
+            uniqueTokens.toString(),
+            hash
+        ).joinToString(separator = ",", postfix = "@#@")
```

Comment:
I think this class is already not flexible at all, because it converts to very concrete format needed for Clone Detection Tool, so maybe there is no need. What do you think?

Reply:
Yes, you are right, thanks!